### PR TITLE
add BasicAuthentication for elasticsearch

### PR DESCRIPTION
**Pull request by witskeeper**
for issue #417 BasicAuthentication for elasticsearch in .net core version?

### CODE REVIEW

File: src/Exceptionless.Web/appsettings.Production.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿---
 ConnectionStrings:
-#  Elasticsearch: server=http://localhost:9200
```
Comment (niemyjski) [inline src/Exceptionless.Web/appsettings.Production.yml]: Can we the changes to the connection strings.

File: src/Exceptionless.Job/appsettings.Production.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 ConnectionStrings:
-#  Elasticsearch: server=http://localhost:9200
```
Comment (niemyjski) [inline src/Exceptionless.Job/appsettings.Production.yml]: Can we the changes to the connection strings.

File: src/Exceptionless.Core/Repositories/Configuration/ExceptionlessElasticConfiguration.cs [hunk 3]
```diff
@@ -60,7 +60,9 @@ public sealed class ExceptionlessElasticConfiguration : ElasticConfiguration, IS
             ConfigureSettings(settings);
             foreach (var index in Indexes)
                 index.ConfigureSettings(settings);
-
+            if (!string.IsNullOrEmpty(Options?.UserName)) {
```
Comment (niemyjski) [inline src/Exceptionless.Core/Repositories/Configuration/ExceptionlessElasticConfiguration.cs]: We should change this to as we shouldn't need to null check it and we should verify both are set.

```cs

if (!String.IsNullOrEmpty(_options.UserName) && !String.IsNullOrEmpty(_options.Password))
    settings.BasicAuthentication(_options.UserName, _options.Password);
```

File: src/Exceptionless.Core/Configuration/ElasticsearchOptions.cs [hunk 4]
```diff
@@ -50,6 +53,10 @@ public class ConfigureElasticsearchOptions : IConfigureOptions<ElasticsearchOpti
             options.FieldsLimit = fieldsLimit > 0 ? fieldsLimit : 1500;
 
             options.EnableMapperSizePlugin = pairs.GetValueOrDefault("enable-size-plugin", _appOptions.Value.AppMode != AppMode.Development);
+
+            options.UserName = pairs.GetString("username", "");
+
+            options.Password = pairs.GetString("password", "");
```
Comment (niemyjski) [inline src/Exceptionless.Core/Configuration/ElasticsearchOptions.cs]: We shouldn't need the second parameter as it returns String.Empty if the key isn't set.

File: src/Exceptionless.Core/Repositories/Configuration/ExceptionlessElasticConfiguration.cs (hunk 3 above)
Comment (witskeeper) [reply src/Exceptionless.Core/Repositories/Configuration/ExceptionlessElasticConfiguration.cs]: I think what you mean is this or not ？
```
if (!String.IsNullOrEmpty(_options.Value.UserName) && !String.IsNullOrEmpty(_options.Value.Password))
    settings.BasicAuthentication(_options.Value.UserName, _options.Value.Password);
```

**Comment by witskeeper** [comment]
i will submit pr again

**Pull request closed without merging**